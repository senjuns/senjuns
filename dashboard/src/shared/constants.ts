import {
  MeasurementTypes,
  MeasurementStatisticsTypes,
  TPasswordRule,
} from '../shared/interfaces';

export const MeasurementUnit = {
  CelciusDegree: '°C',
  FahrenheitDegree: '°F',
  Percent: '%',
  KP: 'kP',
  mph: 'mph',
  ppm: 'ppm',
  DLI: 'DLI',
  μMol_m2_s: 'μMol/m2/s',
};

export const NotificationLevel = {
  Nominal: 'nominal',
  Warning: 'warning',
  Error: 'error',
};

export const ScreenSize = {
  xs: 576,
  sm: 768,
  md: 992,
  lg: 1200,
  xl: 1400,
};

export const Colors = {
  brightBlue100: '#60C3E3',
  brightBlue80: '#7FCEE8',
  brightBlue60: '#9FDAEE',
  grey100: '#E5E5E5',
  grey80: '#EAEAEA',
  grey60: '#EFEFEF',
  black: '#000000',
  dark1: '#383838',
  dark7: '#808080',
  dark8: '#B4B4B4',
  dark9: '#CFCFCF',
  green1: '#56C254',
  grey5: '#E0E0E0',
  hb1: '#82879A',
  hb2: '#9BA0B6',
  hb3: '#C3C7D9',
  hb4: '#ECEDF1',
  hb5: '#F0F1F4',
  hb6: '#F8F8FC',
  orange1: '#DD382F',
  orange2: '#E86339',
  orange3: '#F48E42',
  orange4: '#FFB94C',
  white: '#ffffff',
  linearGradientedOrange: 'linear-gradient(90deg, #E86339 0%, #FFB94C 100%)',
};

export const Sizes = {
  xxSmall: 4,
  xSmall: 8,
  small: 12,
  medium: 16,
  large: 20,
  xLarge: 24,
  xxLarge: 32,
};

export const Fonts = {
  h1: { size: 48, style: 'normal', weight: 600 },
  h2: { size: 40, style: 'normal', weight: 600 },
  h3: { size: 36, style: 'normal', weight: 600 },
  h4: { size: 24, style: 'normal', weight: 600 },
  h5: { size: 20, style: 'normal', weight: 600 },
  h6: { size: 18, style: 'normal', weight: 600 },
  body1: { size: 16, style: 'normal', weight: 'normal' },
  body2: { size: 14, style: 'normal', weight: 'normal' },
};

export const MEASUREMENT_OPTION_DETAILS = {
  [MeasurementTypes.Light]: {
    statisticsKey: MeasurementStatisticsTypes.AirTemperature,
    unit: MeasurementUnit.FahrenheitDegree,
  },
  [MeasurementTypes.Temperature]: {
    statisticsKey: MeasurementStatisticsTypes.AirTemperature,
    unit: MeasurementUnit.FahrenheitDegree,
  },
  [MeasurementTypes.Humidity]: {
    statisticsKey: MeasurementStatisticsTypes.RelativeHumidity,
    unit: MeasurementUnit.Percent,
  },
  [MeasurementTypes.LeafVpd]: {
    statisticsKey: MeasurementStatisticsTypes.LeafVpd,
    unit: MeasurementUnit.KP,
  },
  [MeasurementTypes.Vpd]: {
    statisticsKey: MeasurementStatisticsTypes.AirVpd,
    unit: MeasurementUnit.KP,
  },
  [MeasurementTypes.Airflow]: {
    statisticsKey: MeasurementStatisticsTypes.AirFlow,
    unit: MeasurementUnit.mph,
  },
  [MeasurementTypes.Co2]: {
    statisticsKey: MeasurementStatisticsTypes.AirCo2,
    unit: MeasurementUnit.ppm,
  },
  [MeasurementTypes.Ppfd]: {
    statisticsKey: MeasurementStatisticsTypes.Par,
    unit: MeasurementUnit.μMol_m2_s,
  },
  [MeasurementTypes.LeafTemperature]: {
    statisticsKey: MeasurementStatisticsTypes.LeafTemperature,
    unit: MeasurementUnit.FahrenheitDegree,
  },
};

export const APP_URL = {
  login: '/login',
  forgot: '/forgot',
  reset: '/reset',
  notFound: '/not-found',
  resetPassword: (username: string) => `/reset?username=${username}`,
  userHome: '/user/home',
  userAbout: '/user/about',
  home: '/',
  roomDetails: (id: number) => `/room-details/${id}`,
  roomDetailsOverview: (id: number) => `/room-details/${id}/overview`,
  roomDetailsHeatMap: (id: number) => `/room-details/${id}/heat-map`,
  roomDetailsLineChart: (id: number) => `/room-details/${id}/line-chart`,
  roomDetailsPhotoFeed: (id: number) => `/room-details/${id}/photo-feed`,
};

export enum ROOM_DETAIL_TABS {
  OVERVIEW = 'overview-tab',
  LINE_CHART = 'line-chart-tab',
  HEAT_MAP = 'heat-map-tab',
  PHOTO_FEED = 'photo-feed-tab',
}

export const AWS_API_CODE = {
  // will return when username or password incorrect or when user does not exists.
  user_not_found_exception: 'UserNotFoundException',
};

export const PASSWORD_RULES: TPasswordRule[] = [
  {
    label: 'Password must match',
    check: (password: string, confirmPassword: string): boolean =>
      password === confirmPassword,
  },
  {
    label: 'At least 8 characters',
    check: (password: string): boolean => password.length >= 8,
  },
  {
    label: 'Must contain at least one upper case letter',
    check: (password: string): boolean => /[A-Z]/.test(password),
  },
  {
    label: 'Must contain at least one special character',
    check: (password: string): boolean => /[#?!@$%^&*-]/.test(password),
  },
  {
    label: 'Must contain at least one number',
    check: (password: string): boolean => /[0-9]/.test(password),
  },
];

export const FEATURE_ENABLED = 'FEATURE_ENABLED';

export const NULL_VALUE = -1;

export const Z_INDEXES = {
  DROPDOWN: 10001,
};
