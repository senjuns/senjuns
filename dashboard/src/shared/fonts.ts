import { css } from 'styled-components';

export const InterBoldMirage16px = css`
  color: var(--mirage);
  font-family: var(--font-family-inter);
  font-size: var(--font-size-s);
  font-weight: 700;
  font-style: normal;
`;

export const Poppins18 = css`
  font-family: var(--font-family-poppins);
  font-size: var(--font-size-l2);
  letter-spacing: 0;
  font-weight: 400;
  font-style: normal;
`;

export const Poppins22 = css`
  font-family: var(--font-family-roboto);
  font-size: var(--font-size-22px);
  letter-spacing: 0;
  font-weight: 400;
  font-style: normal;
`;

export const Poppins26 = css`
  font-family: var(--font-family-poppins);
  font-size: var(--font-size-xxl2);
  letter-spacing: 0;
  font-weight: 400;
  font-style: normal;
`;

export const RobotoBoldBlack22px = css`
  color: var(--black);
  font-family: var(--font-family-roboto);
  font-size: var(--font-size-22px);
  font-weight: 700;
  font-style: normal;
`;
