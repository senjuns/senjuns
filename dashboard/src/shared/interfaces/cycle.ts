export interface IPlantInfo {
  strainName: string;
  count: number;
}
