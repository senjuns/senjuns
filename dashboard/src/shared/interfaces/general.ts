export interface Size {
  width: number;
  height: number;
}

export enum METRICS_TYPE {
  USER_LOGIN,
  USER_LOGOUT,
}

export enum METRICS_CATEGORY {
  USER = 'User',
}

export enum FEATURE_FLAGS {
  HEATMAP = 'FEATURE_FLAGS_HEATMAP',
  SETTINGS = 'FEATURE_FLAGS_SETTINGS',
  DASHBOARD = 'FEATURE_FLAGS_DASHBOARD',
  LINE_CHART = 'FEATURE_FLAGS_LINE_CHART',
  CALENDAR = 'FEATURE_FLAGS_LINE_CALENDAR',
  NOTIFICATION = 'FEATURE_FLAGS_NOTIFICATION',
  ANCHOR_DATE_TIME = 'FEATURE_FLAGS_ANCHOR_DATE_TIME',
  DEBUG = 'FEATURE_FLAGS_LOG',
  INTERNAL_HEATMAP_TOOL_FLAG = 'FEATURE_INTERNAL_HEATMAP_TOOL_FLAG',
  AIRFLOW_SENSOR_FLAG = 'AIRFLOW_SENSOR_FLAG',
}

export type TPasswordRule = {
  label: string;
  check: (password: string, confirmPassword: string) => boolean;
};

export type ArrayElement<ArrayType extends readonly unknown[]> =
  ArrayType extends readonly (infer ElementType)[] ? ElementType : never;

export type Nullable<T> = T | null;

export type Maybe<T> = T | null | undefined;

export interface IPosition {
  x: number;
  y: number;
}

export interface MetricsMeta {
  email?: string;
}

export interface TimeRange {
  start: Date;
  end: Date;
}
