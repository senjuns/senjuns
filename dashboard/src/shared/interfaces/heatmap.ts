export type HeatMapStylings = {
  low: {
    max: number;
    color: string;
  };
  mediumLow: {
    max: number;
    color: string;
  };
  medium: {
    max: number;
    color: string;
  };
  mediumHigh: {
    color: string;
    max: number;
  };
  high: {
    color: string;
    max: number;
  };
};
