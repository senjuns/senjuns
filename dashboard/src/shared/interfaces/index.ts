export * from './cycle';
export * from './general';
export * from './heatmap';
export * from './measurement';
