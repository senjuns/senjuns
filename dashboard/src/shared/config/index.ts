import Cookies from 'js-cookie';
// import { FEATURE_ENABLED } from '../../shared/constants';
import { FEATURE_FLAGS } from '../../shared/interfaces';
// @ts-ignore Needs to be ignored as it is dynamically created.
import config from './config';

const defaultConfig: any = config;

if (Cookies.get(FEATURE_FLAGS.DEBUG) === 'FEATURE_ENABLED') {
  console.log(`process.env: ${JSON.stringify(process.env, null, 2)}`);
  console.log(`defaultConfig: ${JSON.stringify(defaultConfig, null, 2)}`);
}

export default defaultConfig;
