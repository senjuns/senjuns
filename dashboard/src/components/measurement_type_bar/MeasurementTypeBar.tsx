import React from 'react';
import styled from 'styled-components';
import { ReactComponent as AirflowIcon } from '../../assets/svg/airflow-icon.svg';
import { ReactComponent as Co2Icon } from '../../assets/svg/co2-icon.svg';
import { ReactComponent as HumidityIcon } from '../../assets/svg/humidity-icon.svg';
import { ReactComponent as LeafTemperatureIcon } from '../../assets/svg/leaf-temperature-icon.svg';
import { ReactComponent as LeafVpdIcon } from '../../assets/svg/leaf-vpd-icon.svg';
import { ReactComponent as PpfdIcon } from '../../assets/svg/ppfd-icon.svg';
import { ReactComponent as TemperatureIcon } from '../../assets/svg/temperature-icon.svg';
import { ReactComponent as VpdIcon } from '../../assets/svg/vpd-icon.svg';
import { Colors } from '../../shared/constants';
import { MeasurementTypes } from '../../shared/interfaces';

interface MeasurementOptionBarProps {
  airflowSensorFlag: boolean;
  value: MeasurementTypes;
  onOptionChange: (option: MeasurementTypes) => void;
}

const MeasurementTypeBar: React.FC<MeasurementOptionBarProps> = ({
  airflowSensorFlag,
  value,
  onOptionChange,
}) => {
  const MEASUREMENT_OPTIONS = [
    {
      type: MeasurementTypes.Temperature,
      Icon: TemperatureIcon,
      enabled: true,
    },
    { type: MeasurementTypes.Humidity, Icon: HumidityIcon, enabled: true },
    { type: MeasurementTypes.LeafVpd, Icon: LeafVpdIcon, enabled: true },
    { type: MeasurementTypes.Vpd, Icon: VpdIcon, enabled: true },
    { type: MeasurementTypes.Co2, Icon: Co2Icon, enabled: true },
    {
      type: MeasurementTypes.Airflow,
      Icon: AirflowIcon,
      enabled: airflowSensorFlag,
    },
    { type: MeasurementTypes.Ppfd, Icon: PpfdIcon, enabled: true },
    {
      type: MeasurementTypes.LeafTemperature,
      Icon: LeafTemperatureIcon,
      enabled: true,
    },
  ];

  return (
    <Container>
      {MEASUREMENT_OPTIONS.map(
        (option) =>
          option.enabled && (
            <OptionItem
              key={option.type}
              selected={value === option.type}
              onClick={() => onOptionChange(option.type)}
            >
              <option.Icon />
            </OptionItem>
          )
      )}
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  overflow: auto;
  box-sizing: border-box;

  &::-webkit-scrollbar {
    display: none;
  }
`;

interface OptionItemProps {
  selected: boolean;
}

const OptionItem = styled.div<OptionItemProps>`
  width: 50px;
  min-width: 50px;
  height: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
  border: ${(props) =>
    props.selected ? `2px solid ${Colors.orange3}` : 'none'};
  border-radius: 10px;
  box-sizing: border-box;
`;

export default MeasurementTypeBar;
