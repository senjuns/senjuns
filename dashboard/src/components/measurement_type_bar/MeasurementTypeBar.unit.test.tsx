import * as ShallowRenderer from 'react-test-renderer/shallow';
import MeasurementTypeBar from './MeasurementTypeBar';
import { MeasurementTypes } from 'shared/interfaces';

describe('<MeasurementCard />', () => {
  it('renders correctly.', async () => {
    const handleOptionChange = jest.fn();
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <MeasurementTypeBar
        onOptionChange={handleOptionChange}
        value={MeasurementTypes.Temperature}
      />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
