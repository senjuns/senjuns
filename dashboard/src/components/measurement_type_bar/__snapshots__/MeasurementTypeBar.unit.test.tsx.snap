// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeasurementCard /> renders correctly. 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
    selected={true}
  >
    <ForwardRef(SvgTemperatureIcon) />
  </styled.div>
  <styled.div
    onClick={[Function]}
    selected={false}
  >
    <ForwardRef(SvgHumidityIcon) />
  </styled.div>
  <styled.div
    onClick={[Function]}
    selected={false}
  >
    <ForwardRef(SvgLeafVpdIcon) />
  </styled.div>
  <styled.div
    onClick={[Function]}
    selected={false}
  >
    <ForwardRef(SvgVpdIcon) />
  </styled.div>
  <styled.div
    onClick={[Function]}
    selected={false}
  >
    <ForwardRef(SvgCo2Icon) />
  </styled.div>
  <styled.div
    onClick={[Function]}
    selected={false}
  >
    <ForwardRef(SvgAirflowIcon) />
  </styled.div>
  <styled.div
    onClick={[Function]}
    selected={false}
  >
    <ForwardRef(SvgPpfdIcon) />
  </styled.div>
  <styled.div
    onClick={[Function]}
    selected={false}
  >
    <ForwardRef(SvgLeafTemperatureIcon) />
  </styled.div>
</styled.div>
`;
